zrml/parimutuel/src/lib.rs

```diff
@@ -459,16 +520,130 @@ mod pallet {
+                    .map(|asset| (AssetOf::<T>::from(asset), None)),
+            )
+        }
+    }
+
+    impl<T: Config> MarketTransitionApi<MarketIdOf<T>> for Pallet<T> {
+        fn on_activation(market_id: &MarketIdOf<T>) -> ResultWithWeightInfo<DispatchResult> {
+            let market_result = T::MarketCommons::market(market_id);
+
+            let market = match market_result {
+                Ok(market) if market.scoring_rule == ScoringRule::Parimutuel => market,
+                Err(e) => {
+                    return ResultWithWeightInfo::new(Err(e), T::DbWeight::get().reads(1));
+                }
+                _ => {
+                    return ResultWithWeightInfo::new(Ok(()), T::DbWeight::get().reads(1));
+                }
+            };
+
+            for outcome in market.outcome_assets(*market_id) {
+                let admin = Self::pot_account(*market_id);
+                let is_sufficient = true;
+                let min_balance = 1u8;
+                if let Err(e) = T::AssetCreator::create(
+                    outcome.into(),
+                    admin,
+                    is_sufficient,
+                    min_balance.into(),
+                ) {
+                    return ResultWithWeightInfo::new(Err(e), T::WeightInfo::on_activation());
+                }
+            }
+
+            ResultWithWeightInfo::new(Ok(()), T::WeightInfo::on_activation())
+        }
+
+        fn on_resolution(market_id: &MarketIdOf<T>) -> ResultWithWeightInfo<DispatchResult> {
+            let market_result = T::MarketCommons::market(market_id);
+
+            let market = match market_result {
+                Ok(market) if market.scoring_rule == ScoringRule::Parimutuel => market,
+                Err(e) => {
+                    return ResultWithWeightInfo::new(Err(e), T::DbWeight::get().reads(1));
+                }
+                _ => {
+                    return ResultWithWeightInfo::new(Ok(()), T::DbWeight::get().reads(1));
+                }
+            };
+
+            let winning_asset_option = market.resolved_outcome_into_asset(*market_id);
+            let winning_asset = if let Some(winning_asset) = winning_asset_option {
+                winning_asset
+            } else {
+                unreachable_non_terminating!(
+                    winning_asset_option.is_some(),
+                    LOG_TARGET,
+                    "Resolved market with id {:?} does not have a resolved outcome",
+                    market_id,
+                );
+                return ResultWithWeightInfo::new(Ok(()), T::DbWeight::get().reads(1));
+            };
+
+            let outcome_total = T::AssetManager::total_issuance(winning_asset.into());
+            let assets_to_destroy = if outcome_total.is_zero() {
+                Self::get_assets_to_destroy(market_id, &market, |asset| {
+                    T::AssetCreator::total_issuance(asset.into()).is_zero()
+                })
```

Comment:
What exactly do you mean by **unexpected** behavior?
The code does call `managed_destroy_multi` with an empty `BTreeMap` from the perspective of the pallet it does not know how the module responsible for deletion does react to it, but in any case an error is just being logged and accepted as non terminating. Since we know what the `Destroyer` type does in our runtime, we can safely say that it does nothing and return `Ok` when calling `managed_destroy_multi` with an emptry `BTreeMap`.

Reply:
By unexpected behavior I meant that if you check `is_zero()` and you expect that this returns true, you would get false, because it is actually `1u8` and not `0u8`.